### ISSUE TITLE
inconsistent types of MMCIF2Dict data values

### ISSUE THREAD

**Issue by speleo3** [biopython/biopython#1533]
### Setup

`Bio.__version__`: 1.71.dev0

### Expected behaviour

Consistent types for `MMCIF2Dict` data values.

### Actual behaviour

It's unpredictable whether a key may hold `list` or `str` type.

If an item has more than one value, it only can be represented in a `loop_`. But if it has a single value, it could either be in a loop or not in a loop. To my understanding, both representations are equivalent.

### Example

Example of a 3 atom file and a 1 atom file. The former represents the `_chem_comp_atom` items in a loop, the latter doesn't:

```python
from urllib2 import urlopen
from Bio.PDB.MMCIF2Dict import MMCIF2Dict

a = MMCIF2Dict(urlopen('https://files.rcsb.org/ligands/download/HOH.cif'))
b = MMCIF2Dict(urlopen('https://files.rcsb.org/ligands/download/MG.cif'))

print(type(a['_chem_comp_atom.model_Cartn_x']))
# <type 'list'>
print(type(b['_chem_comp_atom.model_Cartn_x']))
# <type 'str'>
```

### Suggested Solutions

1. Store all values as lists
2. Or add a `get_list` or `get_iter` method to `MMCIF2Dict`

**Comment by peterjc** [comment]
CC @JoaoRodrigues @jgreener64 @etal

**Comment by speleo3** [comment]
[gemmi](https://github.com/project-gemmi/gemmi) library has a **find_values** method that always returns a list:
http://gemmi.readthedocs.io/en/latest/cif-parser.html#pair-or-loop

**Comment by jgreener64** [comment]
I follow your logic and guess this is an inconsistency, I don't think I'll have time to look at it now though.

I guess as a workaround use:
```python
if isinstance(val, str):
    ...
elif isinstance(val, list):
    ...
```

Of your suggested solutions, the first seems better long term but is breaking. I suppose the second would be implemented by the following in `MMCIF2Dict`:

```python
def get_list(self, key):
    if isinstance(self[key], str):
        return [self[key]]
    elif isinstance (self[key], list):
        return self[key]
```

This could be a simple-ish PR?

**Comment by JoaoRodrigues** [comment]
@jgreener64 why would make things into a list break, could you give an example? Strings are also indexable, for example. I'd think moving to lists would be the best overall.

**Comment by jgreener64** [comment]
I meant in the sense that existing code of users might break if a new Biopython version returned a list when a string was expected (due to a string being returned in an old Biopython version). But I guess people could just update their code.

**Comment by StefansM** [comment]
I've run into this in the past as well. I agree that always returning a `list` would be better than the current heuristic, which only creates lists when in a `_loop`. Right now, it's not obvious when we will get scalars and when we will get lists, which leads to bugs when edge cases are encountered. For example, I didn't know that some structures have more than one `_exptl.method` listed!

In an ideal world, I think it would be best to parse the dictionary definition and determine which entries are allowed to be repeated. I didn't see anything obvious in the dictionary definition specifying which fields can or cannot be repeated, though. I'm not familiar with the STAR format, so maybe I'm missing something.

**Comment by jgreener64** [comment]
#2186 implements the suggestion to store everything as lists.

**Comment by jgreener64** [comment]
I think this can be closed now as #2186 is merged.
